src/adjoint.jl

```diff
@@ -1,16 +1,39 @@
 using MacroTools
 using MacroTools: @q, combinedef
+using ChainRulesCore: AbstractZero, Zero, DoesNotExist
 
 named(arg) = isexpr(arg, :(::)) && length(arg.args) == 1 ? :($(gensym())::$(arg.args[1])) : arg
 
 typeless(x) = MacroTools.postwalk(x -> isexpr(x, :(::), :kw) ? x.args[1] : x, x)
 isvararg(x) = isexpr(x, :(::)) && namify(x.args[2]) == :Vararg
 
+"""
+    generic2diff(x)
+
+Convert input `x` from the internal Zygote format to the ChainRules differential types.
```

Comment:
Since this will be merged at the same time as Zygote starts intenrally using ChainRules types calling this the "internal Zygote format" seems wrong.
I think the "Legacy ZygoteRules" format is clearer.
```suggestion
Convert input `x` from the legacy ZygoteRules format to the ChainRules differential types.
```